utils/to_string.hh

```diff
@@ -63,25 +61,10 @@ std::ostream& operator<<(std::ostream& os, const print_with_comma<NeedsComma, Pr
 
 namespace std {
 
-template<typename Printable>
-static inline
-sstring
-to_string(const std::vector<Printable>& items) {
-    return "[" + utils::join(", ", items) + "]";
-}
-
-template<typename Printable>
-static inline
-sstring
-to_string(const std::set<Printable>& items) {
-    return "{" + utils::join(", ", items) + "}";
-}
-
-template<typename Printable>
-static inline
+template <std::ranges::range Range>
 sstring
-to_string(const std::unordered_set<Printable>& items) {
-    return "{" + utils::join(", ", items) + "}";
+to_string(const Range& items) {
```

Comment:
All the previous functions had "static inline" on them, here you're missing it.
If this is deliberate, please explain why. If not, please add it.

Reply:
Also, don't you need all the exclusions below here as well? What if someone does to_string() on something which is already a string, are we ok that we get a list of bytes?

Reply:
`static inline` in a header is bad practice. It leads to duplication of the function, in case the compiler decides not to inline it.

Reply:
@avikivity you're probably right, but since this patch changes the practice already in to_string.hh it should be mentioned in the commit message (and please consider if there are any more of them left in to_string.hh).

Reply:
> All the previous functions had "static inline" on them, here you're missing it. If this is deliberate, please explain why. If not, please add it.

Templates in the header file don't need the `static inline`, which complicates things as Avi said above.

Reply:
> Also, don't you need all the exclusions below here as well? What if someone does to_string() on something which is already a string, are we ok that we get a list of bytes?

That's why I needed to exclude:
```
+       !std::convertible_to<Range, std::string>
+    && !std::convertible_to<Range, std::string_view>
+    && !std::convertible_to<Range, bytes>
+    && !std::convertible_to<Range, bytes_view>
+    && !std::same_as<Range, boost::unit_test::basic_cstring<const char>>
```
(and `#include <boost/test/utils/basic_cstring/basic_cstring_fwd.hpp>`)).

Otherwise, strings are printed as a range of characters (e.g. "foo" will be printed as `{ 'f', 'o', 'o' }`)
If you insist about not including test header files, I'll try to do that in the test library or find some other solution.

regrading to_string(string), I dropped the `std::to_string` definitions in utils/to_string.hh altogether in the following patches, in favor of fmt::to_string, so they shouldn't matter.

Reply:
For example, without this, `BOOST_REQUIRE(!sv.empty());` prints the following when running with --log all:
```
test/boost/string_format_test.cc(66): info: check {!, s, v, ., e, m, p, t, y, (, )} has passed
```